### DISCUSSION TITLE
How to use internal motor encoders (IMEs) built-in V5 smart motor?

### DISCUSSION THREAD

**Discussion by Williaml21** [LemLib/LemLib#152] [Q&A] [answered]
I'm using the latest LemLib V0.5.0 Release Candidate 7 for my autonomous development. 
In the example code as shown below, it seems assuming to use rotation horizontal tracking wheel for odometry calculation. 

`// tracking wheels
// horizontal tracking wheel encoder. Rotation sensor, port 15, reversed (negative signs don't work due to a pros bug)
pros::Rotation horizontalEnc(15, true);
// horizontal tracking wheel. 2.75" diameter, 3.7" offset, back of the robot (negative)
lemlib::TrackingWheel horizontal(&horizontalEnc, lemlib::Omniwheel::NEW_275, -3.7);`

And in the sensor structure, that's the one being passed in, along with IMU sensor.

I'm using V5 smart motor which has built-in internal motor encoders (IMEs). This would be different from the tracking wheels. I'm not planning to use any tracking wheels, but just IME + IMU. So my question is how do I configure IME sensor and pass its data into the sensor structure to be used. Also, does IMU takes priority over IME for PID control and error correction?

Thanks,
William

**Comment by Raptorly1** [comment, marked as answer]
Hi William! There is no need to explicitly configure IMEs. You can simply have all the values in the `OdomSensors` except for the IMU as `nullptr`
here is an example:
```cpp
pros::Imu imu(2);
lemlib::OdomSensors sensors(
                            nullptr, // vertical tracking wheel 1
                            nullptr, // vertical tracking wheel 2
                            nullptr, // horizontal tracking wheel 1
                            nullptr, // horizontal tracking wheel 2
                            &imu // inertial sensor
);
```

As for your 2nd question, 
> does IMU takes priority over IME for PID control and error correction?

yes it does.

**Reply by Williaml21** [reply]
Thank for your prompt response. Just 10 mins ago, when I looked into the source code of chassis.cpp, I realized that I could use nullptr for all tracking wheels in the constructor, just as you suggested! And it works!

I have other questions, but will create another discussion topic!
